tofhir-engine/src/main/scala/io/tofhir/engine/model/FhirMappingResult.scala

```diff
@@ -73,20 +104,41 @@ case class FhirMappingResult(
     // create a new MapMarker using the marker map
     val marker: MapMarker = new MapMarker("marker", markerMap)
     // add mappedResource to the marker map if error code is INVALID_RESOURCE
-    if (mappedResource.isDefined && error.get.code == FhirMappingErrorCodes.INVALID_RESOURCE)
-      markerMap.put("mappedResource", mappedResource.get)
+    if (mappedFhirResource.isDefined && error.get.code == FhirMappingErrorCodes.INVALID_RESOURCE)
+      markerMap.put("mappedResource", mappedFhirResource.get.mappedResource.get)
     marker
   }
-
 }
 
+/**
+ * Stores the results of mapping i.e. generated FHIR resources for a source input.
+ * It groups the mapped FHIR resources generated from the input data, allowing for easy access to
+ * the output for a specific input row.
+ *
+ * In addition to the common fields from [[BaseFhirMappingResult]], it includes a collection of mapped
+ * FHIR resources, which represent the outcome of the mapping process for each input source.
+ *
+ * @param mappedFhirResources List of mapped FHIR resources. This is used to group mapped resources by input row.
+ */
+case class FhirMappingResultsForInput(
```

Comment:
Can you please extend the description for this class?

FhirMappingResultForInput class represents a the result from a single source row, that is each source row corresponds to one FhirMappingResultForInput instance. If the mapping is 1-to-many, mappedFhirResources field includes all the created FHIR resources from the source row.